cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_misc.hpp

```diff
@@ -0,0 +1,213 @@
+        own_t res;
+        if constexpr (check_mask_flag(bs_list::min, List)) {
+            res.rmin_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::max, List)) {
+            res.rmax_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if (check_mask_flag(bs_list::sum, List) ||
+            (deffered_fin && check_mask_flag(bs_list::mean | sum2cent_based_stat, List))) {
+            res.rsum_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            res.rsum2_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if (check_mask_flag(bs_list::sum2cent, List) ||
+            (deffered_fin &&
+             check_mask_flag(bs_list::varc | bs_list::stdev | bs_list::vart, List))) {
+            res.rsum2cent_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::mean, List)) {
+            res.rmean_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::sorm, List)) {
+            res.rsorm_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::varc, List)) {
+            res.rvarc_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::stdev, List)) {
+            res.rstdev_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+        if constexpr (check_mask_flag(bs_list::vart, List)) {
+            res.rvart_ = pr::ndarray<Float, 1>::empty(q, { count }, alloc::device);
+        }
+
+        return res;
+    }
+
+    auto& get_min() const {
+        return rmin_;
+    }
+    auto& get_max() const {
+        return rmax_;
+    }
+    auto& get_sum() const {
+        return rsum_;
+    }
+    auto& get_sum2() const {
+        return rsum2_;
+    }
+    auto& get_sum2cent() const {
+        return rsum2cent_;
+    }
+    auto& get_mean() const {
+        return rmean_;
+    }
+    auto& get_sorm() const {
+        return rsorm_;
+    }
+    auto& get_varc() const {
+        return rvarc_;
+    }
+    auto& get_stdev() const {
+        return rstdev_;
+    }
+    auto& get_vart() const {
+        return rvart_;
+    }
+
+private:
+    ndresult() = default;
+
+    pr::ndarray<Float, 1> rmin_;
+    pr::ndarray<Float, 1> rmax_;
+    pr::ndarray<Float, 1> rsum_;
+    pr::ndarray<Float, 1> rsum2_;
+    pr::ndarray<Float, 1> rsum2cent_;
+    pr::ndarray<Float, 1> rmean_;
+    pr::ndarray<Float, 1> rsorm_;
+    pr::ndarray<Float, 1> rvarc_;
+    pr::ndarray<Float, 1> rstdev_;
+    pr::ndarray<Float, 1> rvart_;
+};
+
+template <typename Float, bs_list List>
+class ndbuffer {
```

Comment:
not a nd object too

Reply:
It is collection of 1-dimensional ndarrays, therefore named it with nd prefix

Reply:
what is your suggestion? Can suggest local_result local_buffer, is it ok?

Reply:
renamed as local_result and local_buffer_list